File: applicatie/src/main/scala/com/vdzon/administratie/model/Rekening.scala
package com.vdzon.administratie.model

import java.time.LocalDate

import org.bson.types.ObjectId
import org.mongodb.morphia.annotations.{Entity, Id}

import scala.annotation.meta.field

@Entity("rekening") case class Rekening(
                                         @(Id @field) uuid: String,
                                         rekeningNummer: String = null,
                                         factuurNummer: String = null,
                                         naam: String = null,
                                         omschrijving: String = null,
                                         rekeningDate: LocalDate = null,
                                         bedragExBtw: Double = 0,
                                         bedragIncBtw: Double = 0,
                                         btw: Double = 0,
                                         maandenAfschrijving: Int = 0
                                       ) {

  def this() = this(null)


}

File: common/src/main/java/com/vdzon/administratie/model/Rekening.kt
package com.vdzon.administratie.model

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.mongodb.morphia.annotations.Entity
import org.mongodb.morphia.annotations.Id

import java.math.BigDecimal
import java.time.LocalDate

@Entity("rekening")
class Rekening (
        @Id
    val uuid: String = "",
        val rekeningNummer: String = "",
        val factuurNummer: String = "",
        val naam: String = "",
        val omschrijving: String = "",
        @JsonSerialize(using = LocalDateSerializer::class)
    @JsonDeserialize(using = LocalDateDeserializer::class)
    val rekeningDate: LocalDate = LocalDate.now(),
        val bedragExBtw: BigDecimal = BigDecimal.ZERO,
        val bedragIncBtw: BigDecimal = BigDecimal.ZERO,
        val btw: BigDecimal = BigDecimal.ZERO,
        var maandenAfschrijving: Int = 0){
}
